hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/client/HoodieFlinkWriteClient.java

```diff
@@ -111,7 +124,23 @@ public void bootstrap(Option<Map<String, String>> extraMetadata) {
         getTableAndInitCtx(WriteOperationType.UPSERT, instantTime);
     table.validateUpsertSchema();
     preWrite(instantTime, WriteOperationType.UPSERT);
-    HoodieWriteMetadata<List<WriteStatus>> result = table.upsert(context, instantTime, records);
+    final HoodieRecord<T> record = records.get(0);
```

Comment:
why use first record' bucket type?  if first record is `I` and the second one is `U`, th code will use `FlinkCreateHandle` to process all records

Reply:
Yes, that is the logic we want, because we try to merge records into small buckets.

Reply:
So we alse need to deduplicate the record in query stage?